### ISSUE TITLE
can't accept any setting because of hostname checking

### ISSUE THREAD

**Issue by lfarkas** [xoseperez/espurna#880]
in any settings i always got this error message and can't accept any valid parameters:
```
Hostname cannot be empty and may only contain the ASCII letters ('A' through 'Z' and 'a' through 'z'), the digits '0' through '9', and the hyphen ('-')! They can neither start or end with an hyphen.
```
the reason for this that the default hostname is ESPURNA_XXXXXX, where XXXXXX is last 3 octets of chipID (actually 6).
but the hostname check in the code/html/custom.js validateForm do NOT allow underscore "_". So after that if i try to change anything than it's wall with tha above error because of the general/hostname by default will be ESPURNA_3727CF which is not valid according to RFC.
IMHO the best fix would be to change the default hostname in espurna/code/espurna/utils.ino:
```
void setDefaultHostname() {
    if (strlen(HOSTNAME) > 0) {
        setSetting("hostname", HOSTNAME);
    } else {
        setSetting("hostname", getIdentifier());
    }
}
```
to something better eg. replace _ to - of the getIdentifier() and set it as hostname.

**Comment by lfarkas** [comment]
anyway it should have to be allowed that the hostname be FQDN too which is no not possible since "." are not allowed in the hostname.

**Comment by wildwiz** [comment]
about this, let's remember that the underscore is not allowed in host names, but it actually is in domain names, according to
* https://stackoverflow.com/questions/2180465/can-domain-name-subdomains-have-an-underscore-in-it
* https://tools.ietf.org/html/rfc2181#section-11
* https://tools.ietf.org/html/rfc1034

**Comment by xoseperez** [comment]
This has been fixed with @mcspr #889 PR
